### PR TITLE
Add vscode-debugger sample

### PR DESCRIPTION
**Pull request by lorensr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,7 +126,9 @@ and you'll be given the list of sample options.
 #### Production APIs
 
 - [**Production Build**](./production): Build code in advance for faster Worker startup times.
-- [**Debugging**](https://docs.temporal.io/application-development/debugging#debug): The [replay-history](./replay-history) sample shows how to retrieve Workflow Event History and debug it using the `runReplayHistory` Worker API ([video](https://youtu.be/fN5bIL7wc5M)).
+- [**Debugging**](https://docs.temporal.io/application-development/debugging#debug):
+  - The [replay-history](./replay-history) sample shows how to retrieve Workflow Event History and debug it using the `runReplayHistory` Worker API ([video](https://youtu.be/fN5bIL7wc5M)).
```
Comment (bergundy) [inline README.md:130]: I would just remove it at this point and (if there's a reference to it from YT) note in the video description that the sample has been replaced by this newer and shinier one.

File: vscode-debugger/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# VS Code Debugger
+
+Use our [VS Code plugin](https://marketplace.visualstudio.com/items?itemName=temporal-technologies.temporalio) to debug a running or completed Workflow Execution. See this blog post for a walkthrough:
```
Comment (bergundy) [inline vscode-debugger/README.md:3]: Wanna record a video too?

Comment (bergundy) [reply vscode-debugger/README.md:3]: Can just be screen capture, but I guess that might go on the extension README.

File: vscode-debugger/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Client } from '@temporalio/client';
+import { nanoid } from 'nanoid';
+import wait from 'waait';
```
Comment (bergundy) [inline vscode-debugger/src/client.ts:3]: If we don't care about node 14 you can use
```
import { setTimeout } from 'node:timers/promises';
```

File: vscode-debugger/src/client.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Client } from '@temporalio/client';
+import { nanoid } from 'nanoid';
```
Comment (bergundy) [inline vscode-debugger/src/client.ts:2]: And we can just use `crypto.randomUUID()` to remove this dependency.

**Review by bergundy — APPROVED** [review]
Thanks! ❤️

File: vscode-debugger/README.md (hunk 2 above)
Comment (lorensr) [reply vscode-debugger/README.md:3]: Will link video from post

File: vscode-debugger/src/client.ts (hunk 3 above)
Comment (lorensr) [reply vscode-debugger/src/client.ts:3]: i like keeping 14 support in samples as long as SDK supports it

**Pull request merged**